### ISSUE TITLE
[i2c_device] graphic shows Clock is only stretched at the start of a READ/WRITE when ACQ FIFO is full.

### ISSUE THREAD

**Issue by drewmacrae** [lowRISC/opentitan#16158]
https://docs.opentitan.org/hw/ip/i2c/doc/I2C_state_diagram_target.svg Shows no way for clock to be stretched if the acq_fifo is filled within an I2C READ/WRITE

Ideally the clock would stretch whenever the FIFO is full, but if we leave this as described in the picture we would be able to overflow the FIFO and fire the interrupt for that one here:https://github.com/lowRISC/opentitan/issues/15226

**Comment by tjaychen** [comment]
yeah good catch, this seems like a pretty serious issue, and is somewhat related to #15227.
I'll need to have a look.

**Comment by tjaychen** [comment]
i am fixing this in the design and will push a PR soon, but I probably won't fix the documentation in time.
I'm going to punt the documentation fix to M3 for now, but rest assured the design issue will be addressed.

**Comment by tjaychen** [comment]
will be fixed by #16298, which is a companion documentation fix to a slate of clock stretching issues.
